server/src/error.rs

```diff
@@ -20,20 +56,54 @@ impl IntoResponse for CheeseError {
     type BodyError = <Self::Body as axum::body::HttpBody>::Error;
 
     fn into_response(self) -> axum::http::Response<Self::Body> {
-        let bb = match self {
-            Self::Mongo(e) => {
-                event!(Level::ERROR, "MongoDB error: {:?}", e);
-                Self::Body::from("MongoDB did a fuckywucky")
-            }
-            Self::Oid(e) => {
-                event!(Level::ERROR, "ObjectID error: {:?}", e);
-                Self::Body::from("ObjectID did a fuckywucky")
-            }
-        };
-
-        Response::builder()
-            .status(StatusCode::INTERNAL_SERVER_ERROR)
-            .body(bb)
-            .unwrap()
+        match self {
+            Self::Mongo(e) => response_error!(e, internal server error),
+            Self::Oid(e) => response_error!(e, internal server error),
+            CheeseError::Login(l) => match l {
+                e @ LoginError::InvalidCredentials => {
+                    response_error!(e, "unauthorized", StatusCode::UNAUTHORIZED)
+                }
+                LoginError::Database(e) => response_error!(e, internal server error),
+                LoginError::Bcrypt(e) => response_error!(e, internal server error),
+                e @ LoginError::UserWithoutId => response_error!(e, internal server error),
```

Comment:
What's the work flow that causes this to happen? This shouldn't be possible

Reply:
I just don't want the server to crash in these cases. But it can't normally happen unless we have a bug